### PR TITLE
Ability to pass `groupcount` parameter for depthwise and groupwise convolutions

### PR DESCRIPTION
**Pull request by arhik**
~~With the change in #521,~~ We need an api to set groupcount. I am proposing this as a draft and I am planning to make corresponding changes in NNlib. This will break existing NNlib interface since `groupcount` is an extra argument to `ConvDesc`. 

Please feel free to comment on this once I have one of these check boxes complete.

- [x] Add `groupcount` to NNlib ConvDims as a common Descriptor and have seperate constructs in Fluxml.

### CODE REVIEW

File: src/dnn/compat.jl [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Compatibility shims until users upgrade to new NNlib format
 function conv!(y::CuArray{T}, x::CuArray{T}, w::CuArray{T}; pad=0, stride=1, flipkernel=0, dilation=1, kwargs...) where {T<:CUDNNFloat}
-    cdims = DenseConvDims(x, w; padding=pad, stride=stride, flipkernel=flipkernel, dilation=dilation)
```
Comment (arhik) [inline src/dnn/compat.jl]: I need help with these shims.

**Comment by MikeInnes** [comment]
Would be great to have a look from @staticfloat for NNlib compatibility.

**Comment by arhik** [comment]
> Would be great to have a look from @staticfloat for NNlib compatibility.

@staticfloat NNlib related changes are in https://github.com/FluxML/NNlib.jl/pull/146#issue-347196058

These are complete set of changes to test:
Pkg.add(PackageSpec(url="https://github.com/arhik/Zygote.jl.git#groupwise"))
Pkg.add(PackageSpec(url="https://github.com/arhik/CuArrays.jl.git#groupcount"))
Pkg.add(PackageSpec(url="https://github.com/arhik/NNlib.jl.git#common_convdims"))
Pkg.add(PackageSpec(url="https://github.com/arhik/Flux.jl.git#groupwiseconv"))

I have few test cases ready. I will push them whenever I can.

Please observe the branche names after hashes.

**Comment by denizyuret** [comment]
@maleadt any progress on this?

**Comment by maleadt** [comment]
No, I'm not working on any of this.

**Comment by denizyuret** [comment]
@DhairyaLGandhi, is this ball in your court? We need group convolutions.

**Comment by k8lion** [comment]
Any progress? As others have said on the [related issue thread on Flux.jl](https://github.com/FluxML/Flux.jl/issues/459), my research also relies on depthwise convolutions on GPU.

**Comment by denizyuret** [comment]
I implemented this in the latest master.

**Comment by k8lion** [comment]
Great, so now looks NNlib needs to be modified in order for `Flux.DepthwiseConv` to be correctly evaluated on GPU?

**Comment by denizyuret** [comment]
Correct. Or use Knet.

**Comment by DhairyaLGandhi** [comment]
Now that's on cuda master, we should push the same change on NNlib as well.

**Comment by arhik** [comment]
Sorry I didnt have GPU hardware to work and test any of this after pull request. You can follow the changes I made in NNlib, Zygote and Flux if you agree with https://github.com/FluxML/NNlib.jl/pull/146#issue-347196058 . I am not upto date with Zygote, NNlib and Flux development to judge how easy to translate my changes  ...

Edit:
Just checked the diffs. They are easy to translate. Also just to want you : I remember I didn't focus on CPU API. I was only focussed on GPU.

Any one interested in this issue can  also leverage my  mobilenet implementation here. https://github.com/arhik/Flux_mobilenet.git

Cool that this is in limelight. Cant wait to see groupwise convolutions in Flux.

**Pull request closed without merging**